lib/perl/Genome/Model/Build/Command/Abandon.pm

```diff
@@ -52,12 +53,23 @@ sub execute {
     for my $build (@builds) {
         $self->_total_command_count($self->_total_command_count + 1);
         my $transaction = UR::Context::Transaction->begin();
-        my $successful = eval { $build->abandon($self->header_text, $self->body_text) };
-        if ($successful and $transaction->commit) {
-            $self->status_message( "Successfully abandoned build (" . $build->__display_name__ . ")." );
-        }
-        else {
-            $self->append_error($build->__display_name__, "Failed to abandon build: $@.");
+
+        my $successful = try {
+            $build->abandon($self->header_text, $self->body_text);
+        } catch {
+            $self->append_error($build->__display_name__, "Failed to abandon build: $_.");
+            $transaction->rollback;
+        };
+
+        if ($successful) {
+            if ($transaction->commit) {
+                $self->successfully_abandoned_callback($build);
+            } else {
+                $self->append_error($build->__display_name__, "Failed to commit transaction to abandon build");
+                $transaction->rollback;
+            }
+        } else {
+            $self->append_error($build->__display_name__, "Failed to abandon build for some reason.");
             $transaction->rollback;
```

Comment:
I think this is a second call to `rollback()` in the case where `abandon()` dies which I believe will cause an exception.  I think it would also append two errors in that case.